src/lib/Graphs/IO/NodeSpecJSON.ts

```diff
@@ -1,20 +1,19 @@
-export type LinkJSON = { node: number; socket: string };
+import { NodeCategory } from '../../Nodes/NodeCategory';
 
 export type InputSocketSpecJSON = {
-  name: string,
-  valueType: string,
-  defaultValue: string;
+  name: string;
+  valueType: 'flow' |'string'|'number'|'boolean',
+  defaultValue?: string | number | boolean
 }
+
 export type OutputSocketSpecJSON = {
-  name: string,
-  valueType: string,
+  name: string;
+  valueType: 'flow' |'string'|'number'|'boolean'
```

Comment:
This should be a string so it is compatible with the new ValueTypeRegistry approach.